src/main/kotlin/io/snyk/plugin/ui/jcef/ApplyFixHandler.kt

```diff
@@ -0,0 +1,138 @@
+package io.snyk.plugin.ui.jcef
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.ui.jcef.JBCefBrowserBase
+import com.intellij.ui.jcef.JBCefJSQuery
+import io.snyk.plugin.DiffPatcher
+import io.snyk.plugin.toVirtualFile
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+import org.jetbrains.concurrency.runAsync
+import io.snyk.plugin.ui.SnykBalloonNotificationHelper
+import java.io.IOException
+
+
+data class DiffPatch(
+    val originalFile: String,
+    val fixedFile: String,
+    val hunks: List<Hunk>
+)
+
+data class Hunk(
+    val startLineOriginal: Int,
+    val numLinesOriginal: Int,
+    val startLineFixed: Int,
+    val numLinesFixed: Int,
+    val changes: List<Change>
+)
+
+sealed class Change {
+    data class Addition(val line: String) : Change()
+    data class Deletion(val line: String) : Change()
+    data class Context(val line: String) : Change()  // Unchanged line for context
+}
+
+class ApplyFixHandler(private val project: Project) {
+
+    private val enableDebug = Logger.getInstance("Snyk Language Server").isDebugEnabled
+    private val enableTrace = Logger.getInstance("Snyk Language Server").isTraceEnabled
+    private val logger = Logger.getInstance(this::class.java)
```

Comment:
Just do sth like this:
```
val logger = Logger.getInstance(this::class.java).apply {
            // tie log level to language server log level
            if (languageServerWrapper.logger.isDebugEnabled) this.setLevel(LogLevel.DEBUG)
            if (languageServerWrapper.logger.isTraceEnabled) this.setLevel(LogLevel.TRACE)
        }
```
This sets the log level of the logger, so it only writes out messages that are on this level or above. In the code, you'd then decide which information is necessary on info, debug, trace level to manage logging verbosity adequately.

Reply:
Fixed